### ISSUE TITLE
tests doesn't work on windows

### ISSUE THREAD

**Issue by dptd** [dptd/erno#6]
Visual Studio solution is created correctly via CMake. erno executable and ernolib builds fine. Tests however doesn't work. It looks like some issues with the runtime library but I have not solved it yet.

For now ernolib is a static library, gtest is static too. When trying to build algorithmsUnitTests I got 68 errors.
```
Error	LNK2005	"protected: __thiscall std::basic_streambuf<char,struct std::char_traits<char> >::basic_streambuf<char,struct std::char_traits<char> >(void)" (??0?$basic_streambuf@DU?$char_traits@D@std@@@std@@IAE@XZ) already defined in gmock_main.lib(gtest-all.obj)	algorithmsUnitTests	D:\Developer\erno\build\msvcprtd.lib(MSVCP140D.dll)	1	
Error	LNK2005	"public: virtual __thiscall std::basic_streambuf<char,struct std::char_traits<char> >::~basic_streambuf<char,struct std::char_traits<char> >(void)" (??1?$basic_streambuf@DU?$char_traits@D@std@@@std@@UAE@XZ) already defined in gmock_main.lib(gtest-all.obj)	algorithmsUnitTests	D:\Developer\erno\build\msvcprtd.lib(MSVCP140D.dll)	1	
Error	LNK2005	"protected: char * __thiscall std::basic_streambuf<char,struct std::char_traits<char> >::eback(void)const " (?eback@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IBEPADXZ) already defined in gmock_main.lib(gtest-all.obj)	algorithmsUnitTests	D:\Developer\erno\build\msvcprtd.lib(MSVCP140D.dll)	1	
Error	LNK2005	"protected: char * __thiscall std::basic_streambuf<char,struct std::char_traits<char> >::gptr(void)const " (?gptr@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IBEPADXZ) already defined in gmock_main.lib(gtest-all.obj)	algorithmsUnitTests	D:\Developer\erno\build\msvcprtd.lib(MSVCP140D.dll)	1	
```

**Comment by dptd** [comment]
I see that in generated liberno solution Runtime Library is set to `Multi-threaded Debug DLL (/MDd)` but in gtest it is set to `Multi-threaded Debug (/MTd)`. According to [this](https://stackoverflow.com/questions/15527401/google-tests-in-a-dll) post it should be the same in both projects.

The question is how to set it from CMake.

**Comment by dptd** [comment]
fixed in 92eae76242c575264591f492b3944ab1122752ff
